internal/kafka/internal/services/kafka.go

```diff
@@ -567,25 +568,54 @@ func (k *kafkaService) DeprovisionKafkaForUsers(users []string) *errors.ServiceE
 	return nil
 }
 
-func (k *kafkaService) DeprovisionExpiredKafkas(kafkaAgeInHours int) *errors.ServiceError {
-	dbConn := k.connectionFactory.New().
-		Model(&dbapi.KafkaRequest{}).
-		Where("instance_type IN (?)", kafkaDeletionInstanceTypes).
-		Where("created_at <= ?", time.Now().Add(-1*time.Duration(kafkaAgeInHours)*time.Hour)).
-		Where("status NOT IN (?)", kafkaDeletionStatuses)
+func (k *kafkaService) DeprovisionExpiredKafkas() *errors.ServiceError {
+	dbConn := k.connectionFactory.New().Model(&dbapi.KafkaRequest{}).Session(&gorm.Session{})
 
-	db := dbConn.Update("status", constants2.KafkaRequestStatusDeprovision)
+	var existingKafkaRequests []dbapi.KafkaRequest
+	db := dbConn.Where("instance_type IN (?)", kafkaDeletionInstanceTypes).
```

Comment:
At the moment the implementation only retrieves `developer` instances because we know that those are the only ones that have expiration.
In theory we shouldn't do this as they could have expiration too. It is kind of an "optimization". Do you think we should just make it fully generic without filtering by the instance type?

Reply:
That makes sense :+1: I will implement it

Reply:
I updated the PR to include the additional proposed logic here